client/helpers.go

```diff
@@ -79,38 +80,34 @@ func changeTargetsDelegation(repo *tuf.Repo, c changelist.Change) error {
 		if err != nil {
 			return err
 		}
-		r, _, err := repo.GetDelegation(c.Scope())
-		if _, ok := err.(data.ErrNoSuchRole); err != nil && !ok {
-			// error that wasn't ErrNoSuchRole
+		_, err = repo.GetDelegationRole(c.Scope())
+		if _, ok := err.(data.ErrInvalidRole); err != nil && !ok {
+			// error that wasn't ErrInvalidRole
 			return err
 		}
+
 		if err == nil {
-			// role existed, attempt to merge paths and keys
-			if err := r.AddPaths(td.AddPaths); err != nil {
-				return err
-			}
-			return repo.UpdateDelegations(r, td.AddKeys)
-		}
-		// create brand new role
-		r, err = td.ToNewRole(c.Scope())
-		if err != nil {
-			return err
+			// role existed, merge paths and keys to this copy of the delegation role
+			addTD := changelist.TufDelegation{AddPaths: td.AddPaths, AddKeys: td.AddKeys}
+			return repo.UpdateDelegations(c.Scope(), addTD)
```

Comment:
I see what you're saying -- my only concern is that it will be really verbose to handle all of the args unless we break down UpdateDelegations, and breaking it down might further fragment our validation or make `changeTargetsDelegation` more complicated.  It also seems like the changelist package is currently pretty light on imports.  I'm happy to give it a shot if you think it'd be better to pass the changelist as unpacked arguments, though I'm thinking it might be better to keep the changelist itself as a compact argument to keep the API cleaner.   cc @cyli for additional input